packages/editor/src/hooks/block-settings-inspector-toggle.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { __unstableBlockSettingsMenuFirstItem as BlockSettingsMenuFirstItem } from '@wordpress/block-editor';
+import { MenuItem } from '@wordpress/components';
+import { store as interfaceStore } from '@wordpress/interface';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __, isRTL } from '@wordpress/i18n';
+import { drawerLeft, drawerRight } from '@wordpress/icons';
+
+/**
+ * Override the default edit UI to include a new block inspector control for
+ * assigning a partial syncing controls to supported blocks in the pattern editor.
+ * Currently, only the `core/paragraph` block is supported.
+ *
+ * @param {Component} BlockEdit Original component.
+ *
+ * @return {Component} Wrapped component.
+ */
+const withBlockToolbar = createHigherOrderComponent(
+	( BlockEdit ) => ( props ) => {
+		return (
+			<>
+				<BlockEdit key="edit" { ...props } />
+				{ props.isSelected && <BlockInspectorToggle /> }
+			</>
+		);
+	},
+	'withBlockToolbar'
+);
+
+const BlockInspectorToggle = () => {
+	const { enableComplementaryArea } = useDispatch( interfaceStore );
+
+	const isOpen = useSelect(
+		( select ) =>
+			select( interfaceStore ).getActiveComplementaryArea( 'core' ) ===
+			'edit-post/block',
+		[]
+	);
+
+	return (
+		<BlockSettingsMenuFirstItem>
+			<MenuItem
+				onClick={ () => {
+					enableComplementaryArea( 'core', 'edit-post/block' );
+				} }
+				icon={ isRTL() ? drawerLeft : drawerRight }
+				disabled={ isOpen }
+			>
+				{ __( 'Open Block settings' ) }
+			</MenuItem>
+		</BlockSettingsMenuFirstItem>
+	);
+};
+
+addFilter(
+	'editor.BlockEdit',
+	'core/editor/with-block-settings-inspector-toggle',
+	withBlockToolbar
+);
```

Comment:
I'm unsure as to whether a hook is the best option here. We could try adding the component here

https://github.com/WordPress/gutenberg/blob/8b4ca3437a086bf7dae71c81561f3e90b1dc0e67/packages/editor/src/components/provider/index.js#L301-L318

This would avoid it running on each block.

Reply:
@youknowriad I would value your perspective as to whether menu items that apply to all blocks in the Editor (via the `@wordpress/editor` package) are better included as a component in the Provider? Or is using a hook (as currently in this PR) suitable?

I've noted my concerns around performance of the hooks approach which is why I am suggesting this alternative.